File: src/test/resources/org/springframework/data/hadoop/scripting/basic-script.js
importPackage(java.util);
importPackage(org.apache.hadoop.fs);

println("Home dir is " + fs.homeDirectory)
println("Work dir is " + fs.workingDirectory)
println("/user exists " + fs.exists("/user"))

name = UUID.randomUUID().toString()
scriptName = "src/test/resources/test.properties"
fs.copyFromLocalFile(scriptName, name)
// use the shell
dir = "script-dir/"
if (!fsh.test(dir)) {
   fsh.mkdir(dir); fsh.cp(name, dir); fsh.chmodr(700, dir)
   println("File content is " + fsh.cat(dir + name))
}

println(fsh.ls(dir))
fsh.rmr(dir)

// return the file length 
fs.getLength(name)

File: spring-hadoop-test/src/test/resources/org/springframework/data/hadoop/scripting/basic-script.rb
require 'java'
include org.apache.hadoop.fs

puts "Home dir is " + $fs.homeDirectory.to_s
puts "Work dir is " + $fs.workingDirectory.to_s
puts "/user exists " + $fs.exists("/user").to_s

name = java.util.UUID.randomUUID().to_s
scriptName = "src/test/resources/test.properties"
$fs.copyFromLocalFile(scriptName, name)
puts Path.new(name).makeQualified($fs)

# use the shell
dir = "script-dir/"
if !$fsh.test(dir)
	$fsh.mkdir(dir)
	$fsh.cp(name, dir)
	$fsh.chmodr(700, dir)
	print "File content is " + $fsh.cat(dir + name).to_s
end

print $fsh.ls(dir).to_s
$fsh.rmr(dir)

# return the file length
$fs.getLength(name)